lib/practice_exercises.rb

```diff
@@ -1,13 +1,44 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) because the while how long the loop runs, depends on the lenght of the string
+# Space Complexity: 0(1)  
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  # raise NotImplementedError, "Not implemented yet"
+  string = string.downcase.gsub(/[^a-z^A-Z]/,"")
+
+  i = 0
+  j = string.length - 1
+  while i < j
+    if string[i] == string[j]
+      i += 1
+      j -= 1
+    else
+      return false
+    end
+end
+return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n) because of the length of the string??? <-- this one is questionable. 
+# Space Complexity: 0(1) because no array is being made
 def longest_prefix(strings)
```

Comment:
Actually you are making a string called "prefix" so your space complexity is O(n)